aea/helpers/dialogue/base.py

```diff
@@ -697,6 +696,7 @@ def update(self, message: Message) -> Optional[Dialogue]:
             dialogue = self.get_dialogue(message)
 
         if dialogue is not None:
+            message.counterparty = dialogue.dialogue_label.dialogue_opponent_addr
```

Comment:
Do we have a check in here that guarantees the counterparty is not overwritten and throws an error on inconsistencies?

Reply:
Done